R/zchunk_L241.en_newtech_nonco2.R

```diff
@@ -1,107 +1,282 @@
+                 by = c("region", "supplysector", "subsector", "stub.technology", "Non.CO2")) %>%
+      na.omit ->
+      L241.nonco2_max_reduction
+
+
+    # Steepness of emission reduction
     #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # Assign the steepness of emission reduction based on energy technologies to all regions.
+    A51.steepness %>%
+      gather(Non.CO2, value, -supplysector, -subsector, -stub.technology) %>%
+      repeat_add_columns(tibble(region = GCAM_region_names[["region"]])) ->
+      L241.steepness
+
+    # Select closest future max reduction emission coefficients.
+    L241.nonco2_max_reduction %>%
+      select(region, supplysector, subsector, stub.technology, Non.CO2, year) %>%
+      filter(year == min(FUTURE_YEARS)) %>%
+      mutate(ctrl.name = "GDP_control") ->
+      L241.nonco2_steepness
+
+    # Combine closet future max reduction emission coefficients with the assumed steepness
+    # of emission reductions into one data frame.
+    L241.steepness %>%
+      select(steepness = value, region, supplysector, subsector, stub.technology, Non.CO2) %>%
+      # Use left_join here because we expect there to be NAs.
+      left_join(L241.nonco2_steepness, by = c("region", "supplysector", "subsector",  "stub.technology", "Non.CO2")) %>%
+      na.omit ->
+      L241.nonco2_steepness
+
+    # Rename SO2 to regional SO2.
+    L241.nonco2_tech_coeff <- rename_SO2(L241.nonco2_tech_coeff, Emiss_A_regions, FALSE)
+    L241.nonco2_max_reduction <- rename_SO2(L241.nonco2_max_reduction, Emiss_A_regions, FALSE)
+    L241.nonco2_steepness <- rename_SO2(L241.nonco2_steepness, Emiss_A_regions, FALSE)
+
+
+    # TODO: better way to handle this, probably these technologies should pull from historical data
+    # Ensure only regions that have first gen biofuels get read in
+    # See GitHub issue #650
+
+    # Use the may be historic new technology emission coefficients to replace the emission
+    # coefficients in the min model base year by a supplysecotr, subsector, stub.technology identifier
+    # called sector_tech_id.
+    A41.tech_coeff %>%
+      select(supplysector, subsector, stub.technology, may.be.historic) %>%
+      na.omit %>%
+      unite(sector_tech_id, supplysector, subsector, stub.technology, remove = FALSE) ->
+      L241.maybe_historic
+
+    # Replace the new technology emission coefficients in the model base year with the may be historic
+    # values from the technology assumption file by the unique supplysector, subsector,
```

Comment:
"with the may be historic values"?

Reply:
I was going to just fix and merge this but have to agree–it's a confusing comment, and the `may.be.historic` flag isn't documented in the `A41.tech_coeff.csv` header either. Otherwise, this PR is good to go.

Reply:
Sorry about the vague commenting, I have no idea what the `may.be.historic` values are or how to better describe what is going on here.  The old ds had a the following comment `Allow some future techs that may be in the historical years` would that make more sense?

Reply:
I'd say ask, probably @pkyle but perhaps someone else, for clarification.

Reply:
OK I think I understand this now...these are technologies that are available in most or all regions in future years, but only in the historical years in some cases. For example, coal to liquids exists in the base years in South Africa, whereas most everywhere else it's only allowed in future years. By default, these "newtech" nonco2 parameters start in the first future model time period (2015), but because some of these "newtechs" can be in the historical data in some regions and years, their nonco2 data are instead started (generally, in all regions) in the first model base year (1975).
That said, I don't see any purpose to these steps; all of the nonco2 data could have just been assigned to 1975 for all technologies, with no consequences b/c where output is zero, emissions are also zero. This step also doesn't reduce the file size of the XML, b/c each tech's nonco2 data is only read into one year, and the model internally copies it forward to all subsequent years. That said, my first guess is that the motivation for this step was reducing the size of the XML files. My second guess is that these technologies were intended to be excluded from getting emissions coefs in the base years, because they weren't considered in the mappings from inventory emissions data to GCAM technologies. So, having non-zero historical coefs means that they will produce emissions, so then the model output won't match the emissions inventory data. But again, what's done here would achieve the opposite of that outcome, so really I don't know!

Reply:
Thanks, Page! Kalyn, can you distill that down to a useful comment? If not ask @pkyle to say the same thing but using fewer words :)

Reply:
I think so. This is the part of the chunk that I think the comment 
``` {r} 
# TODO: better way to handle this, probably these technologies should pull from historical data
# Ensure only regions that have first gen biofuels get read in
# See GitHub issue #650
 ```
is referring to. @pkyle & @bpbond do we want to figure out a better way to do this now or wait?

Reply:
No, no need now. Let's wrap this up–making sure the comment here is clear and in A41 too–and then get 'er merged.